### PR TITLE
Simplify `DescendantRebaser`

### PR DESCRIPTION
**Pull request by martinvonz**
This makes the `rebase_one()` step of `DescendantRebaser` simpler and safer. References are now updated after all commits have been rebased.

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: lib/src/repo.rs [hunk 1]
```diff
@@ -857,9 +857,15 @@ impl MutableRepo {
     /// The `rebase_descendants` logic will rebase the descendants of `old_id`
     /// to become the descendants of parent(s) of `old_id`. Any branches at
     /// `old_id` would be moved to the parent(s) of `old_id` as well.
+    /// TODO: Propagate errors from commit lookup or take a Commit as argument.
```
Comment (yuja) [inline lib/src/repo.rs:860]: nit: `// TODO` (no-doc comment)

File: lib/src/rewrite.rs [hunk 2]
```diff
@@ -516,9 +503,23 @@ impl<'settings, 'repo> DescendantRebaser<'settings, 'repo> {
...
+        let mut heads_to_add: HashSet<_> = heads_to_add_expression
+            .evaluate_programmatic(self.mut_repo)
+            .unwrap()
+            .iter()
+            .collect();
+
         let mut heads_to_remove = vec![];
         for old_parent_id in self.mut_repo.parent_mapping.keys() {
-            self.heads_to_add.remove(old_parent_id);
+            heads_to_add.remove(old_parent_id);
```
Comment (yuja) [inline lib/src/rewrite.rs:517]: nit: maybe we can remove this because `old_commits_expression` is excluded?
(might be better to do in a separate commit)

**Review by yuja — APPROVED** [review]
Nice. I don't have much expertise, but looks good to me.

File: lib/src/repo.rs (hunk 1 above)
Comment (martinvonz) [reply lib/src/repo.rs]: Done.

File: lib/src/rewrite.rs (hunk 2 above)
Comment (martinvonz) [reply lib/src/rewrite.rs:517]: True. I think we can make some further simplifications here too. We should also rewrite `update_references()` and `update_wc_commits()`. I'll leave all those for another PR.
